### PR TITLE
TRAN-100: Add modal to add new Supplier or edit already existing one

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
**Add modal to add new Supplier or edit already existing one**

**Implemented solution detail:**
TRAN-100

### CODE REVIEW

File: src/pages/menuUpload/supplierMaster/SupplierMasterPage.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import { PostSupplierMasterRequestPayload, PutSupplierMasterRequestPayload } from '../../../redux/types/supplierMasterType';
+import { ISupplierMaster } from '../../../models/supplierMaster/ISupplierMasterType';
+
+function SupplierMasterPage() {
+  const [displayAddModal, setDisplayAddModal] = useState(false);
+  const [displayEditModal, setDisplayEditModal] = useState(false);
+  const [objectToEdit, setObjectToEdit] = useState<ISupplierMaster>({
+    id: undefined,
+    name: '',
+
```
Comment (sniedzielski) [inline src/pages/menuUpload/supplierMaster/SupplierMasterPage.tsx]: is this empty space necessary?

File: src/components/forms/supplierMaster/SupplierMasterForm.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+import Input from '../../../shared/inputs/input';
+import SubmitButton from '../../../shared/buttons/SubmitButton';
+import CancelButton from '../../../shared/buttons/CancelButton';
+import ValidationError from '../../shared/ValidationError';
+import DeleteButton from '../../../shared/buttons/DeleteButton';
+import SupplierFormType from './types';
+
+function SupplierMasterForm({
+  onSubmit, onCancel, title, submitButtonText, initialFormValue, mode,
+}: SupplierFormType) {
```
Comment (sniedzielski) [inline src/components/forms/supplierMaster/SupplierMasterForm.tsx:14]: Can you use ManualFormType ? (you can find in form/types.tsx)

**Review by sniedzielski — COMMENTED** [review]
LGTM , left two comments. Please refer to the one with 'SupplierFormType'. Please also add hover on edit/delete icons. Thx in advance!

**Review by sniedzielski — APPROVED** [review]

File: src/components/forms/supplierMaster/SupplierMasterForm.tsx (hunk 2 above)
Comment (KarolinaKopacz) [reply src/components/forms/supplierMaster/SupplierMasterForm.tsx:14]: Every form cannot have the same type, because of different types in redux

**Pull request merged**